### PR TITLE
Api check

### PR DESCRIPTION
**Pull request by shirimaasen**
https://trello.com/c/hSW3YSy3/124-backend-bulk-test-and-optimization

### CODE REVIEW

File: api/int_news/views.py [hunk 1]
```diff
@@ -2,55 +2,86 @@
...
+            if serializer.is_valid():
+                serializer.save()
+
+        return Response(data={"200": "OK"}, status=status.HTTP_200_OK)
+
+    def delete(self, request, *args, **kwargs):
+        news = NewsInt.objects.all()
+        news.delete()
+        return Response({'Status': 'OK'},status=status.HTTP_200_OK)
+
```
Comment (Martinfree) [inline api/int_news/views.py]: @Green5tar please check why it's not working

File: api/authentication/serializers.py [hunk 2]
```diff
@@ -290,17 +288,14 @@ class Meta(object):
 
 
 class BulkUpdateStudentSerializer(serializers.ModelSerializer):
-    faculty = serializers.CharField(max_length=128)
```
Comment (Azrael-git) [inline api/authentication/serializers.py:293]: Need to be a FacultySerializer field

File: api/authentication/serializers.py [hunk 3]
```diff
@@ -277,10 +278,7 @@ class Meta(object):
 
 
 class UpdateStudentSerializer(serializers.ModelSerializer):
-    faculty = serializers.CharField(max_length=128)
```
Comment (Azrael-git) [inline api/authentication/serializers.py:280]: Need to be a FacultySerializer field

File: api/authentication/serializers.py [hunk 4]
```diff
@@ -290,17 +288,14 @@ class Meta(object):
 
 
 class BulkUpdateStudentSerializer(serializers.ModelSerializer):
-    faculty = serializers.CharField(max_length=128)
-    profession = serializers.CharField(max_length=128)
```
Comment (Azrael-git) [inline api/authentication/serializers.py:294]: Need to be a ProfessionSerializer field

**Review by Azrael-git — CHANGES_REQUESTED** [review]
Please make changes, as i show you in my last commit in this PR. After that I will check again.

Good luck!

File: api/authentication/serializers.py [hunk 5]
```diff
@@ -251,6 +251,7 @@ class Meta(object):
         fields = ('faculty',)
 
     def save(self, user, data):
+        print(data)
```
Comment (Azrael-git) [inline api/authentication/serializers.py]: Delete print

File: api/authentication/serializers.py [hunk 6]
```diff
@@ -286,15 +289,22 @@ class Meta(object):
...
+        try:
+            name = data.get('faculty').get('name')
+            f = Faculty.objects.get(name=name)
+            print(f)
+            user.faculty = f
 
-class BulkUpdateStudentSerializer(serializers.ModelSerializer):
+            name2 = data.get('profession').get('name')
+            f2 = Profession.objects.get(name=name2)
+            print(f2)
```
Comment (Azrael-git) [inline api/authentication/serializers.py]: Delete print

File: api/authentication/views.py [hunk 7]
```diff
@@ -280,19 +279,19 @@ def post(self, request, *args, **kwargs):
...
         queryset = Student.objects.all()
 
         for user in list(queryset):
-            serializer = BulkUpdateStudentSerializer(user,  data=request.data)
-            if serializer.is_valid():
-                serializer.save()
-                return Response(data={"200": "OK"}, status=status.HTTP_200_OK)
+            serializer = UpdateStudentSerializer(user,  data=request.data)
+            if serializer.is_valid(raise_exception=True):
+                print(request.data)
```
Comment (Azrael-git) [inline api/authentication/views.py]: Delete print

**Review by Azrael-git — CHANGES_REQUESTED** [review]

File: api/authentication/views.py [hunk 8]
```diff
@@ -54,23 +53,23 @@ def get(self, request, *args, **kwargs):
...
-    
+
     def post(self, request, *args, **kwargs):
         return Response(status=status.HTTP_405_METHOD_NOT_ALLOWED)
-    
-    @permissions(["IsModeratorUser", "IsUser"])
+
+    @permissions(["IsModeratorUser", "IsUser"],"kwargs")
     def put(self, request, *args, **kwargs):
         number = args[0]
```
Comment (Azrael-git) [inline api/authentication/views.py]: In decorator you're using kwargs, so args cause error. Need to be fixed. You have to use only kwargs variable

File: api/authentication/views.py [hunk 9]
```diff
@@ -54,23 +53,23 @@ def get(self, request, *args, **kwargs):
...
         if(serializer.is_valid()):
             serializer.save()
-        
+
         return Response(status=status.HTTP_200_OK)
-        
-    @permissions(["IsModeratorUser", "IsUser"])
+
+    @permissions(["IsModeratorUser", "IsUser"],"kwargs")
     def delete(self, request, *args, **kwargs):
```
Comment (Azrael-git) [inline api/authentication/views.py:71]: In decorator you're using kwargs, so args cause error. Need to be fixed. You have to use only kwargs variable

File: api/authentication/views.py [hunk 10]
```diff
@@ -122,25 +121,25 @@ def delete(self, request, *args, **kwargs):
...
     lookup_field = 'id'
     serializer_class = UserSerializer
-    queryset = User.objects.all()
+    queryset = User.objects.none()
     permission_classes = [AllowAny,]
 
-    @permissions(["IsStaffUser", "IsUser"])
+    @permissions(["IsStaffUser", "IsUser"],"kwargs")
     def get(self, request, *args, **kwargs):
-        number = args[0].get('id')
```
Comment (Azrael-git) [inline api/authentication/views.py:130]: In decorator you're using kwargs, so args cause error. Need to be fixed. You have to use only kwargs variable

**Review by Azrael-git — CHANGES_REQUESTED** [review]

**Comment by Azrael-git** [comment]
And please, check all views.py for same mistakes

File: api/authentication/views.py [hunk 11]
```diff
@@ -83,7 +82,7 @@ class UsersAPIView(ListAPIView, CreateAPIView):
     permission_classes = [AllowAny,]
     serializer_class = UserSerializer
 
-    @permission("IsStaffUser")
+    #@permission("IsStaffUser")
```
Comment (Azrael-git) [inline api/authentication/views.py]: Why have you commented this?

**Review by Azrael-git — CHANGES_REQUESTED** [review]

**Comment by Azrael-git** [comment]
I think good this time

**Review by Azrael-git — APPROVED** [review]

**Pull request merged**